### PR TITLE
feat(component): `<PricingFeatures />`

### PR DESCRIPTION
**Pull request by pbortnick**
🎟️ [Asana Task]()
🔍 [Preview Link](https://react-components-git-pbpricing-features-hashicorp.vercel.app/components/pricingfeatures)

---

Alignment of sticky tiers for each possible amount of tiers/columns:

1tier
2tiers
3tiers
4tiers
5tiers

## Description

Add `<PricingFeatures />` component package to be used in new pricing pages project

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

**Review by alexcarpenter — COMMENTED** [review]
First bit of feedback. Looking good so far!

File: packages/pricing-features/components/table/index.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                      scope="col"
+                      colSpan={colIdx === 0 && colCount > 3 ? 2 : 1}
+                    >
+                      {col.length ? (
+                        <div className={s.colHeaderText}>
+                          <span>{col}</span>
+                        </div>
+                      ) : (
+                        <VisuallyHidden>
+                          <span>Title Column</span>
```
[the remark this answers is not in the document]
Comment (pbortnick) [reply packages/pricing-features/components/table/index.tsx:60]: I thought someone commented suggesting using `VisuallyHidden` here but I can't find it and may have been mistaken. We can alternatively use an empty div like in `<StickyTiers />`

File: packages/pricing-features/components/sticky-tiers/index.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  return (
+    <div
+      className={classNames(s.stickyTiers, isVisible && s.isVisible)}
+      style={
+        {
+          '--grid-template-columns':
+            tiersCount > 3
+              ? `2fr repeat(${tiersCount}, 1fr)`
+              : `repeat(${tiersCount + 1}, 1fr)`,
+          '--col-gap': tiersCount === 2 ? '34px' : '22px',
```
[the remark this answers is not in the document]
Comment (pbortnick) [reply packages/pricing-features/components/sticky-tiers/index.tsx]: Done [here](https://github.com/hashicorp/react-components/pull/689/commits/d0e2e2133617564ba288905f0bd5916fdd8f0d19#diff-6ec06ae5fa98f5ca1284aeeea299c8980230f9434eee7cf40b96eb3045e260ddR21-R26)

File: packages/pricing-features/components/sticky-tiers/index.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      aria-hidden={true}
+    >
+      <div className={s.inner}>
+        <div />
+        {tiers.map(({ title, cta }) => (
+          <div key={title} className={s.tier}>
+            <p className={s.tierName}>{title}</p>
+            <div className={s.cta}>
+              <Link href={cta.url}>
+                <a onClick={cta.onClick} tabIndex={-1}>
```
[the remark this answers is not in the document]
Comment (pbortnick) [reply packages/pricing-features/components/sticky-tiers/index.tsx:52]: Added! [view here](https://github.com/hashicorp/react-components/pull/689/commits/d0e2e2133617564ba288905f0bd5916fdd8f0d19#diff-6ec06ae5fa98f5ca1284aeeea299c8980230f9434eee7cf40b96eb3045e260ddR51)

File: packages/pricing-features/components/sticky-tiers/style.module.css [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    background: var(--white);
+    align-items: center;
+    grid-template-columns: var(--grid-template-columns);
+    padding: 16px 0 24px;
+    display: grid;
+    width: 100%;
+    border-bottom-left-radius: 8px;
+    border-bottom-right-radius: 8px;
+    border: 1px solid rgba(0, 0, 0, 0.2);
+    filter: drop-shadow(0 4px 19px #c1c3c8);
```
[the remark this answers is not in the document]
Comment (pbortnick) [reply packages/pricing-features/components/sticky-tiers/style.module.css]: [done](https://github.com/hashicorp/react-components/pull/689/commits/173459bf19e18f91291d0fc6201747d49e4aef2f)

File: packages/pricing-features/components/sticky-tiers/style.module.css [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  & .inner {
+    background: var(--white);
+    align-items: center;
+    grid-template-columns: var(--grid-template-columns);
+    padding: 16px 0 24px;
+    display: grid;
+    width: 100%;
+    border-bottom-left-radius: 8px;
+    border-bottom-right-radius: 8px;
+    border: 1px solid rgba(0, 0, 0, 0.2);
```
[the remark this answers is not in the document]
Comment (pbortnick) [reply packages/pricing-features/components/sticky-tiers/style.module.css]: [done](https://github.com/hashicorp/react-components/pull/689/commits/173459bf19e18f91291d0fc6201747d49e4aef2f)

File: packages/pricing-features/components/table/style.module.css [hunk 6]
```diff
@@ -0,0 +1,187 @@
+.pricingTable {
+  background-color: var(--gray-6);
```
[the remark this answers is not in the document]
Comment (pbortnick) [reply packages/pricing-features/components/table/style.module.css]: [done](https://github.com/hashicorp/react-components/pull/689/commits/43d774552294d353911142c74d8843970af96167)

File: packages/pricing-features/components/table/index.tsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+                      scope="row"
+                      colSpan={colCount > 3 ? 2 : 1}
+                      className={s.rowHeader}
+                    >
+                      <div className={s.rowHeaderInner}>
+                        {isCollapsible && (
+                          <button
+                            className={s.icon}
+                            aria-controls={ariaControls.join(' ')}
+                            aria-expanded="false"
```
[the remark this answers is not in the document]
Comment (pbortnick) [reply packages/pricing-features/components/table/index.tsx]: [done](https://github.com/hashicorp/react-components/pull/689/commits/0a825afde2323971cbeee5b161086d8f63d4d51c). Note - edited the `cellIds` so there are fewer instances of `IDs used in ARIA and labels must be unique`

**Review by alexcarpenter — APPROVED** [review]
👍🏼

**Pull request merged**